openedx/core/djangoapps/user_authn/docs/decisions/0001_simplify_ratelimiting.rst

```diff
@@ -0,0 +1,106 @@
+Simplify Login and Other Rate Limiting
+======================================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+edx-platform currently uses multiple different ratelimiting tools currently
+which can lead to confusion and difficulty understand how endpoints are
+secured.  Consider the following case study in how our login endpoints are
+currently rate limitied.
+
+1st party vs 3rd party login
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+edx-platform allows for both 1st party auth, where you provide the LMS with
+your credentials and it gives you back some session tokens, and also 3rd party
+auth, in which you are directed to a 3rd party to authenticate and then
+redirected to the LMS with a token from that third party which is exchanged for
+1st party(LMS) session tokens.
+
+Login User View
+~~~~~~~~~~~~~~~
+
+The ``login_user`` view in ``views/login.py`` is called for both 1st and 3rd
+party login flows.  In the 3rd party login flow, it's called as the callback
+when redirected from the 3rd party login back to the LMS.
+
+Currently this view is accessed through two different endpoints.  It is
+currently a pass-through call for the ``/api/user/v1/account/login_session``
+using a DRF view.  It is also called directly at ``/login_ajax``
+
+Rate limiting Logins
+~~~~~~~~~~~~~~~~~~~~
+
+Currently there are five different rate limiting implementations in use as a
+part of the login flow.
+
+* `django-ratelimit`_ - A 3rd party rate limiting library that allows you to
```

Comment:
@robrap converting this to a list-table will take me more time than I want to spend on this right now so leaving it as is for now.